particula/aerosol_dynamics/parcel.py

```diff
@@ -48,43 +38,68 @@ def name(self) -> str:
     def add_particle(self, particle_object):
         """Adds a particle to the parcel.
         Parameters:
-            particle_data (list): The radius of the particle.
+            particle_object (object)
         """
-        self._particle_data.append(particle_object)
+        self._particle_data = np.append(self._particle_data, particle_object)
 
     def create_and_add_particle(
-        self, name: str, radius, density=1e3, charge=0
+        self, name: str, radius, density=1e3 * u.kg / u.m**3,
+        charge=0 * u.dimensionless
     ):
         """creats and then Adds a particle to the parcel.
         Parameters:
-            particle_data (list): The radius of the particle.
+            name    (str)   [no units],
+            radius  (float) [m]
+        Optional:
+            density (float) [kg/m**3]       default = 1e3 [kg/m**3],
+            charge  (int)   [dimensionless] default = 0 [dimensionless]
         """
-        self._particle_data.append(
+
+        try:  # maybe move to particle class
+            radius.check('m')
+        except AttributeError:
+            print('Please add units to radius. E.g., radius = 150 * u.nm]')
+
+        self.add_particle(
             particle.Particle(name, radius, density, charge)
         )
 
     def create_and_add_list_of_particle(
```

Comment:
When it is a long list like this may better to give each item its own line (and perhaps add a comma at the end for future safety when copying and pasting and/or adding stuff):

```python
    def create_and_add_list_of_particle(
         self, 
         name_of_particles, 
         radius_of_particles,
         density_of_particles=None, 
         charge_of_particles=None,
     ):
```

Reply:
good idea